api/config/v1alpha1/envoygateway_types.go

```diff
@@ -86,6 +91,55 @@ type FileProvider struct {
 	// TODO: Add config as use cases are better understood.
 }
 
+type Extension struct {
+	// Name defines the name to register with for the extension.
+	Name string
+
+	// APIGroups defines the set of K8s api groups the extension will handle.
+	APIGroups []string
+
+	// Service defines the configuration of the extension service that the Envoy
+	// Gateway Control Plane will call through extension hooks.
+	Service *ExtensionService
+}
+
+type ExtensionService struct {
+	// Host define the extension service hostname.
+	Host string `json:"host"`
+
+	// Port defines the port the extension service is exposed on.
+	//
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:default=80
+	Port int32 `json:"port,omitempty"`
+
+	// TLS defines TLS configuration for communication between Envoy Gateway and
+	// the extension service.
+	//
+	// +optional
+	TLS *ExtensionTLS `json:"tls,omitempty"`
```

Comment:
Can we steal something from the Gateway API on this so it is cohesive with that? 

https://pkg.go.dev/sigs.k8s.io/gateway-api@v0.6.0/apis/v1beta1#GatewayTLSConfig

Thoughts?

Reply:
This type is slightly different than the one I wrote. For example https://pkg.go.dev/sigs.k8s.io/gateway-api@v0.6.0/apis/v1beta1#TLSModeType is not necessary and not something I think we want to expose because that exposes too much detail about the service deployment. From the perspective of Envoy Gateway it shouldn't matter if we're going to put an LB in front of our extensions service so why should we care about where the TLS gets terminated?

Additionally `ExtensionTLS` can eventually support loading certs from places other than K8s secrets. Though right now we only support K8s secrets